chapters/functions/programming-principles/reading/README.md

```diff
@@ -1 +1,321 @@
+
+### Details
+
+Once a project reaches more than five lines of code, it should be stored in one or more separate files.
+In order to combine these files a single "source" command file can be created
+
+Typically function definitions are stored in separate files, so one may have several separate files like: `input.r`, `function.r`, `analysis.r`, `output.r`.
+While developing the analysis, the data would only be read once with `source("input.r")`.
+The goal of this practice is to end up with a set of files which are completely self-contained, so one can start with an empty R session and give only the commands like:
+
+* `source("input.r")`
+* `source("function.r")`
+* `source("analysis.r")`
+* `source("output.r")`
+
+### Examples
+
+:::info Example
+
+For a given project `input.r`, `function.r`, `analysis.r` and `output.r` files can be created as below.
+
+* `input.r`
+
+  ```text
+  dat <- read.table("http://notendur.hi.is/~gunnar/kennsla/alsm/data/set115.dat", header=T)
+  ```
+
+* functions.r:
+
+  ```text
+  plotwtle <- function(fnum) {
+  fname<-paste("http://notendur.hi.is/~gunnar/kennsla/alsm/data/set",fnum,".dat",sep="")
+  cat("The URL is",fname,"\n")
+  dat <- read.table(fname,header=T)
+  ttl <- paste("My data set was",fnum)
+  plot(dat$le,dat$osl,main=ttl,xlab="Length(cm)",ylab="Live weight (g)")
+  }
+  ```
+
+* output.r:
+
+  ```text
+  source("functions.r")
+  for(i in 101:150) {
+    fnam<-paste("plot",i,".pdf",sep="")
+    pdf(fnam) plotwtle(i)
+    dev.off()
+  }
+  ```
+
+These files can be executed with source commands as below:
+
+* `source("input.r")`
+* `source("function.r")`
+* `source("output.r")`
+
+:::
+
+## Structuring an R Project
+
+### Details
+
+We already covered how to split code into different functions and linking them together with the help of one executable file that is "sourcing" the others.
+However, when you undertake a larger project, there will be a lot of different data and files and it is very advisable to have a consistent structure throughout the project
+
+A common project layout is to allocate all project files into a folder, something along the lines of:
+
+```text
+/project /data /src /doc /figs (or /out)
+```
+
+Such a structure is quite normal in programming languages such as C, Matlab, and R.
+
+Purpose of the different folders:
+
+- `/data`: Contains all important data to the project, which you will use.
+  This folder should be read-only! No function is allowed to write anything into this folder
+
+- `/src`: (abbreviation for `source(-code)`) Here you will store all the functions that you programmed.
+  You can decide to store the executable function here as well or, alternatively, have that one in the root project folder
+
+- `/doc`: Contains further documentation material about your project.
+  This could be, for example, readme files for other people who use your functions, or the paper you wrote about the project, or the $\latex$ files while you're writing
```

Comment:
```suggestion
  This could be, for example, readme files for other people who use your functions, or the paper you wrote about the project, or the $\LaTeX$ files while you're writing
```